File: src/main/java/book/system/services/RentalService.java
package book.system.services;

import book.system.dto.RentalDTO;
import book.system.models.Book;
import book.system.models.Rental;

import java.util.List;

public interface RentalService {
    List<RentalDTO> findAll();

    RentalDTO create(Book book);

    RentalDTO returnBook(Rental rental);
}

File: front/src/app/services/rental.service.ts
import { Injectable } from '@angular/core';
import { Rental } from '../models/rental';
import { HttpClient } from '@angular/common/http';

@Injectable()
export class RentalService {

  rentalUrl: string;

  constructor(private http: HttpClient) {
    this.rentalUrl = 'http://localhost:8080/api/rent/';
  }

  public findAll() {
    return this.http.get<Rental[]>(this.rentalUrl);
  }

  public create(id: number) {
    return this.http.get<Rental>(this.rentalUrl + id);
  }

  public returnBook(id: number) {
    return this.http.get<Rental>(this.rentalUrl + 'return/' + id);
  }
}
